source/dagger/gameplay/platformer/platformer_controller.cpp

```diff
@@ -45,25 +53,75 @@ void PlatformerControllerSystem::Run()
                 }
             }
 
-            if (char_.isRolling)
+            if (jump == 1 && !char_.isRolling)
             {
-                if (char_.timeRolling < char_.rollingTime)
+                char_.isJumping = true;
+                if (run != 0)
                 {
-                    AnimatorPlay(animator_, "souls_like_knight_character:ROLL");
-                    sprite_.position.x += char_.rollingSpeed * sprite_.scale.x * Engine::DeltaTime();
-                    char_.timeRolling += Engine::DeltaTime();
+                    sprite_.scale.x = run;
                 }
-                else
+            }
+
+            if (char_.isRolling || char_.isJumping)
+            {
+
+                if (char_.isRolling)
+                {
+                    if (run != 0)
+                    {
+                        sprite_.scale.x = run;
+                    }
+                    if (char_.timeRolling < char_.rollingTime)
+                    {
+                        AnimatorPlay(animator_, "souls_like_knight_character:ROLL");
+                        sprite_.position.x += char_.rollingSpeed * sprite_.scale.x * Engine::DeltaTime();
+                        char_.timeRolling += Engine::DeltaTime();
+                    }
+                    else
+                    {
+                        char_.isRolling = false;
+                        char_.timeRolling = 0;
+                    }
+                }
+                else if (char_.isJumping)
                 {
-                    char_.isRolling = false;
-                    char_.timeRolling = 0;
+                    if (run != 0)
+                    {
+                        sprite_.scale.x = run;
+                    }
+                    Float32 speed = CalculateVerticalSpeed(char_.verticalInitialSpeed, char_.gravity, char_.timeJumping);
+                    if (!char_.reachedMax)
+                    {
+                        AnimatorPlay(animator_, "souls_like_knight_character:JUMP");
+                        char_.timeJumping += Engine::DeltaTime();
+                        sprite_.position.y += speed * Engine::DeltaTime();
+                        if (char_.isRunning) sprite_.position.x += char_.speed * sprite_.scale.x * Engine::DeltaTime();
+                        else sprite_.position.x += sprite_.scale.x * Engine::DeltaTime();
```

Comment:
I would add a set of curly braces every time we use if/else in order to maintain readability. 
And maybe use it as (this second part is just my preference):
if (condition)
{
}
instead of:
if (condition) {
}

Reply:
This is the part of the styleguide I need to add, still. If in a situation to use `{}`, try to always use them (we sometimes leave them and then review and get them in later), but always put them in the next line. It's very important for readability. Next year, we'll maybe even have a linter for all of this, to give you useful comments and auto-corrections on the spot... not this one, though :D